stl/inc/ranges

```diff
@@ -704,16 +704,19 @@ namespace ranges {
         /* [[no_unique_address]] */ _Copyable_box<_Ty> _Val{};
     };
 
+    template <class _Ty>
+    single_view(_Ty) -> single_view<_Ty>;
+
     namespace views {
         // VARIABLE views::single
         struct _Single_fn {
             // clang-format off
             template <class _Ty>
             _NODISCARD constexpr auto operator()(_Ty&& _Val) const noexcept(
-                noexcept(single_view{_STD forward<_Ty>(_Val)})) requires requires {
-                single_view{static_cast<_Ty&&>(_Val)};
+                noexcept(single_view<decay_t<_Ty>>(_STD forward<_Ty>(_Val)))) requires requires {
+                single_view<decay_t<_Ty>>(static_cast<_Ty&&>(_Val));

```

Comment:
Conventionally, we ensure that the first formation of a new type in a _requires-expression_ is in a _type-requirement_. The type `single_view<decay_t<_Ty>>`, however, is first formed in this _simple-requirement_. (The intent here is that simpler requirements should result in simpler error messages for users.)